Dockerfile

```diff
@@ -1,32 +1,38 @@
-FROM  debian:9
+FROM  debian:10
 
 ENV TC_DIR     /usr/local/trinitycore
 ENV TC_REPO    git://github.com/TrinityCore/TrinityCore.git
+ENV TC_DB_URL  https://github.com/TrinityCore/TrinityCore/releases/download/TDB335.20051/TDB_full_world_335.20051_2020_05_15.7z
+#TODO Add mechanism to pull latest releases from TC repo
 
 ENV DEBIAN_FRONTEND noninteractive
 
 RUN apt-get update && \
   apt-get install -y \
   git \
+  wget \
+  clang \
   cmake \
   make \
   gcc \
   g++ \
   libmariadbclient-dev \
-  libssl1.0-dev \
+  libssl-dev \
   libbz2-dev \
   libreadline-dev \
   libncurses-dev \
   libboost-all-dev \
-  mysql-server \
+  mariadb-server \
   p7zip \
-  zlib1g \
-  zlib1g-dev \
-  libreadline-dev
+  default-libmysqlclient-dev \
+  p7zip-full
+
+RUN update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+RUN update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang 100
 
 RUN mkdir -p $TC_DIR && \
   cd $TC_DIR && \
-  git clone -b 3.3.5 --depth 1 $TC_REPO
+  git clone -b 3.3.5 $TC_REPO
```

Comment:
Just curious, why exclude `--depth 1`?

Reply:
I have tested with the depth 1 and it does not compile. Just reverting to the command without it it works. So it is kind of mandatory (and clearly correct in the TC installation guide)